lock/src/main/java/com/auth0/android/lock/views/DomainFormView.java

```diff
@@ -0,0 +1,243 @@
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+
+import com.auth0.android.lock.Configuration;
+import com.auth0.android.lock.R;
+import com.auth0.android.lock.events.EnterpriseROLoginEvent;
+import com.auth0.android.lock.events.EnterpriseWebLoginEvent;
+import com.auth0.android.lock.utils.Strategies;
+import com.auth0.android.lock.utils.Strategy;
+import com.squareup.otto.Bus;
+
+import java.util.List;
+
+public class DomainFormView extends FormView {
+
+    private static final String TAG = DomainFormView.class.getSimpleName();
+    private ValidatedInputView emailInput;
+    private ValidatedInputView usernameInput;
+    private ValidatedInputView passwordInput;
+    private Strategy currentStrategy;
+    private String currentUsername;
+    private List<Strategy> filteredEnterpriseStrategies;
+    private ValidatedInputView.DataType usernameEmailValidation;
+    private Button actionButton;
+    private Button goBackBtn;
+
+    public DomainFormView(Context context) {
+        super(context);
+    }
+
+    public DomainFormView(Context context, Bus lockBus, Configuration configuration) {
+        super(context, lockBus, configuration);
+    }
+
+    @Override
+    protected void init(Configuration configuration) {
+        inflate(getContext(), R.layout.com_auth0_lock_domain_form_view, this);
+        filteredEnterpriseStrategies = configuration.getEnterpriseStrategies();
+
+        actionButton = (Button) findViewById(R.id.com_auth0_lock_action_btn);
+        actionButton.setText(R.string.com_auth0_lock_action_login);
+        actionButton.setOnClickListener(this);
+        actionButton.setEnabled(false);
+        passwordInput = (ValidatedInputView) findViewById(R.id.com_auth0_lock_input_password);
+        passwordInput.setVisibility(View.GONE);
+        usernameInput = (ValidatedInputView) findViewById(R.id.com_auth0_lock_input_username);
+        usernameInput.setVisibility(View.GONE);
+        goBackBtn = (Button) findViewById(R.id.com_auth0_lock_back_btn);
+        goBackBtn.setText(R.string.com_auth0_lock_action_go_back);
+        goBackBtn.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                emailInput.clearInput();
+                resetDomain();
+            }
+        });
+        goBackBtn.setVisibility(GONE);
+
+        emailInput = (ValidatedInputView) findViewById(R.id.com_auth0_lock_input_username_email);
+        emailInput.setDataType(ValidatedInputView.DataType.EMAIL);
+        emailInput.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                String text = s.toString();
```

Comment:
Let's move the logic of domain matching to a different object and properly test it (otherwise it won't be tested at all). Maybe an object that for a domain returns the matched connection or nothing